src/constants.rs

```diff
@@ -53,7 +56,7 @@ pub const DEFAULT_L2_MAINNET_SERVICES: &[L2MainnetService] = &[
     L2MainnetService::PublicNode,
 ];
 
-pub const CONTENT_TYPE_HEADER: &str = "Content-Type";
+pub const CONTENT_TYPE_HEADER_LOWERCASE: &str = "content-type";
 pub const CONTENT_TYPE_VALUE: &str = "application/json";
 
 pub const ETH_MAINNET_CHAIN_ID: u64 = 1;
```

Comment:
This might be useful for custom providers, but we could remove it for now. The idea is that if we need to block a problematic hostname or convert this to an allowlist, the logic is fully set up for us to implement this relatively quickly.